include/seqan3/io/sequence_file/input.hpp

```diff
@@ -824,11 +831,21 @@ class sequence_file_input
     file_as_tuple_type columns_buffer;
     //!\}
 
-    //!\brief Path of the file that the stream operates on.
-    std::string file_name;
-
-    //!\brief The stream we are reading from.
-    stream_type stream;
+    /*!\name Stream / file access
+     * \{
+     */
+    //!\brief The type of the internal stream pointers. Allows dynamically setting ownership management.
+    using stream_ptr_t = std::unique_ptr<std::basic_istream<stream_char_type>,
+                                         std::function<void(std::basic_istream<stream_char_type>*)>>;
+    //!\brief Stream deleter that does nothing (no ownership assumed).
+    static constexpr auto stream_deleter_noop     = [] (std::basic_istream<stream_char_type> *) {};
+    //!\brief Stream deleter with default behaviour (ownership assumed).
+    static constexpr auto stream_deleter_default  = [] (std::basic_istream<stream_char_type> * ptr) { delete ptr; };
```

Comment:
would it not make sense to pass the deleters to the make_secondary... functions, in order to avoid code duplication? But I don't feel to strong about it as they are basically independent.

Reply:
I thought about this and came to the conclusion that it needlessly clutters the interfaces and likely even result in more LOCs.

Reply:
ok